### PR TITLE
feat: support mybatis page query

### PR DESCRIPTION
**Pull request by YongwuHe**

### CODE REVIEW

File: arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
 import org.apache.ibatis.mapping.MappedStatement;
 import org.apache.ibatis.reflection.Reflector;
 
+import java.util.Map;
+
 public class InternalExecutor {
 
     private static final char KEYHOLDER_SEPARATOR = ';';
     private static final char KEYHOLDER_TYPE_SEPARATOR = ',';
+    private static boolean needSearchPage = false;
```
Comment (mr3) [inline arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java:20]: Suggest renaming to `isIPageClassLoaded`

File: arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java [hunk 2]
```diff
@@ -49,6 +70,45 @@ public static <U> void record(DatabaseExtractor extractor,
         }
     }
 
+    private static void savePage(DatabaseExtractor extractor, IPage<?> page) {
+        extractor.setPage(Serializer.serialize(page));
+    }
+
+    public static <T> IPage<T> findPage(Object parameterObject) {
```
Comment (mr3) [inline arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java:77]: Suggest renaming to `extractPage`

File: arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java [hunk 3]
```diff
@@ -27,8 +45,11 @@ public static MockResult replay(DatabaseExtractor extractor, MappedStatement ms,
         return replayResult;
     }
 
-    public static <U> void record(MappedStatement ms, Object o, String originalSql, U result, Throwable throwable, String methodName) {
-        DatabaseExtractor extractor = createExtractor(ms, originalSql, o, methodName);
+    public static <U> void record(DatabaseExtractor extractor, Object parameterObject, U result, Throwable throwable) {
+        Object page = findPage(parameterObject);
+        if (page != null) {
+            savePage(extractor, (IPage<?>) page);
+        }
```
Comment (mr3) [inline arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java:52]: Suggest use `extractor.setPage(extractPage(parameterObject));`

File: arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java [hunk 4]
```diff
@@ -49,6 +70,45 @@ public static <U> void record(DatabaseExtractor extractor,
...
+                }
+            }
+        }
+        if (parameterObject instanceof IPage) {
+            return (IPage<T>) parameterObject;
+        }
+        return null;
+    }
+
+    private static <T> void restorePage(DatabaseExtractor extractor, IPage<T> originalPage) {
```
Comment (mr3) [inline arex-instrumentation/database/arex-database-mybatis3/src/main/java/io/arex/inst/database/mybatis3/InternalExecutor.java:96]:
```java
private static <T> void restorePage(DatabaseExtractor extractor, Object parameterObject) {
    IPage<T> page = extractPage(parameterObject);
    if (page == null) { return; }
}
```

**Review by mr3 — APPROVED** [review]

**Review by lucas-myx — APPROVED** [review]

**Pull request merged**